src/basic/unit-name.c

```diff
@@ -730,22 +732,32 @@ int unit_name_mangle_with_suffix(
         if (string_is_glob(name) && in_charset(name, VALID_CHARS_GLOB)) {
                 if (flags & UNIT_NAME_MANGLE_GLOB)
                         goto good;
-                log_full(flags & UNIT_NAME_MANGLE_WARN ? LOG_NOTICE : LOG_DEBUG,
+                log_full(warn ? LOG_NOTICE : LOG_DEBUG,
                          "Glob pattern passed%s%s, but globs are not supported for this.",
                          operation ? " " : "", strempty(operation));
                 suggest_escape = false;
         }
 
-        if (is_device_path(name)) {
-                r = unit_name_from_path(name, ".device", ret);
-                if (r >= 0)
-                        return 1;
-                if (r != -EINVAL)
+        if (path_is_absolute(name)) {
+                _cleanup_free_ char *t = NULL;
+                ChaseFlags chase_flags = CHASE_NOFOLLOW | CHASE_NONEXISTENT | CHASE_NO_AUTOFS |
+                        (warn ? CHASE_WARN : 0);
+
+                r = chase(name, NULL, chase_flags, &t, NULL);
+                log_debug_errno(r, "Failed to chase %s: %m", name);
+                if (ERRNO_IS_NEG_RESOURCE(r))
```

Comment:
Spurious logging. Triggers assertion if the file does not exsit, as it returns `0`.

Reply:
Arrggh. Should be fixed now.